xtevent/_eventiv.ado

```diff
@@ -234,13 +234,17 @@ program define _eventiv, rclass
 	loc komit: list uniq komit	
 	
 	* Check that the iv normalization works
-
+	
 	foreach v in `leadivs' `varivs' {
-		qui _regress `v' `included' [`weight'`exp'] if `touse', absorb(`i')
-		if e(r2)==1 {
-			di as err "Instrument is collinear with the included event-time dummies. You may have generated leads of the policy variable and included them in the proxyiv option instead of specifying the lead numbers."
-			exit 301
-		}	
+		foreach evtd in `included'{
+			qui _rmcoll `v' `evtd' if `touse'

```

Comment:
@Constantino-Carreto-Romero This is checking collinearity of each iv against a single event-time dummy at each time, but we want to check collinearity with all the event time dummies at the same time. So I think here you need something like 
```_rmcoll `v' `included`if `touse' ```

If there's collinearity and you still want the error message to single out the collinear variable, you can probably loop over the combinations of incluided using `tuples`.